aten/src/ATen/native/cuda/ROCmWorkaround.cuh

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+namespace rocm { namespace workaround {
+
+template<typename T>
+struct enable_default_constructor {
+  struct { char bytes[sizeof(T)]; } value;
+  __device__ operator T&() {
+    return *reinterpret_cast<T *>(&value);
```

Comment:
I think this needs to be `return *std::launder(reinterpret_cast<T *>(&value));` to ensure the lifetime is valid.

Reply:
`std::launder` seems to be a feature of C++17. I am not sure if it is usable here. Let me try.
https://en.cppreference.com/w/cpp/utility/launder

Reply:
Confirmed that this is causing compiler error